File: Training.Demo.Code.2020.Object.Oriented.Programming/C++/Encapsulation/Camera.cpp
#include<iostream>

using namespace std;
// Data and methods that operate on the data are encapsulated together
class Camera {

private:
	float zoomLevel;
    float exposureLevel;
    int framesPerSecond;
    bool isFlashOn;

public:
	void zoomIn() {
        if (zoomLevel < 4) {
            zoomLevel += 0.1;
        }
    }

    void zoomOut() {
        if(zoomLevel > 0) {
            zoomLevel -= 0.1;
        }
    }

    void setExposureLevel(float exposureLevel) {
        this->exposureLevel = exposureLevel;
    }

    void setFramesPerSecond(int framesPerSecond) {
        this->framesPerSecond = framesPerSecond;
    }

    void toggleFlashState() {
        isFlashOn = !isFlashOn;
    }
};

File: Java/Encapsulation/Camera.java
// Data and methods that operate on the data are encapsulated together
public class Camera {

    private float zoomLevel;
    private float exposureLevel;
    private int framesPerSecond;
    private boolean isFlashOn;

    public void zoomIn() {
        if (zoomLevel < 4) {
            zoomLevel += 0.1;
        }
    }

    public void zoomOut() {
        if(zoomLevel > 0) {
            zoomLevel -= 0.1;
        }
    }

    public void setExposureLevel(float exposureLevel) {
        this.exposureLevel = exposureLevel;
    }

    public void setFramesPerSecond(int framesPerSecond) {
        this.framesPerSecond = framesPerSecond;
    }

    public void toggleFlashState() {
        isFlashOn = !isFlashOn;
    }
}
